src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/InputFallThroughLabeledBreak.java

```diff
@@ -0,0 +1,101 @@
+
+    int method(int a, int b) {
+        top: switch (a) {
+            case 1:
+                switch (b) {
+                    case 10:
+                        break top; // ok
+                    default:
+                        return 11;
+                }
+            case 2:
+                return 2;
+        }
+        outer: outer2: switch (a) {
+            case 1:
+                inner1: inner2: switch (b) {
+                    case 10:
+                        break outer; // ok
+                    default:
+                        return 11;
+                }
+            case 2:
+                inner: for (int i = 0;i < 3;i++) {
+                    break outer2; // ok
+                }
+            case 3:
+                return 2;
+        }
+
+        loop_outer: for(int c = 0; c < 2; c++) {
+            outer:switch (a) {
+                case 1:
+                    inner:inner2:switch (b) {
+                        case 10:
+                            break inner;
+                        default:
+                            return 11;
+                    }
+                case 2: // violation 'Fall through from previous branch of the switch statement.'
+                    inner: inner2: for (int i = 0; i < 3; i++) {
+                        break inner2;
+                    }
+                case 3: // violation 'Fall through from previous branch of the switch statement.'
+                    return 1;
+                case 4:
+                    switch (b) {
+                        case 10:
+                            break outer; // ok
+                        default:
+                            return 11;
+                    }
+                case 5:
+                    inner:{ switch (a) {
+                        case 1: break inner;
+                    }
+                    }
+                case 6: // violation 'Fall through from previous branch of the switch statement.'
+                    inner:
+                    while (true) {
+                        break outer;
+                    }
+                case 7:
+                    inner: while (true) {
+                        break inner;
+                    }
+                case 8:  // violation 'Fall through from previous branch of the switch statement.'
+                    inner:
+                    while (true) {
+                        continue loop_outer;
+                    }
+                case 9:
+                    loop_inner:
+                    while (true) {
+                        continue loop_inner;
+                    }
+                case 10:  // violation 'Fall through from previous branch of the switch statement.'
+                    inner1:
+                    {
+                        switch (a) {
+                            case 1:
+                                break outer; // ok
+                        }
+                    }
+                case 11:
```

Comment:
If we observe, in the ```
Violation lines for /Users/piyushkumarsadangi/Development/checkstyle/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/InputFallThroughLabeledBreak.java differ.
unexpected (2): 37, 95
---
expected      : [50, 54, 68, 77, 87]
but was       : [37, 50, 54, 68, 77, 87, 95]

```

Now, ideally we shouldn't be getting any violation at this line: `case 11:`. As we exit the `outer:switch (a)`. However, we continue to get a violation now, which must not have been the case when the old PR was being authored, because the code base has undergone some changes. Now, the question remains do we revert back to old changes, or have a fusion of both worlds, line-number based approach and AST. I am not sure on this. Please let me know if i have understood right. I will continue to understand the existing code, and maybe a change here and there can still get us no violation at 37 and 95. I hope I didn't complicate with my observations.

Reply:
Yes, 37 and 95 should not be violations.